In [13]:
#General libraries

import numpy as np
import pandas as pd

#Visualization libraries

import matplotlib.pyplot as plt
import seaborn as sns


In [14]:
#Importing data
df = pd.read_csv('D:/Prog_project_summer/assets/Salary_Data_Based_country_and_race.csv')


In [15]:
#Reviewing the first five data
df.head()

Unnamed: 0   Age  Gender Education Level          Job Title  \
0           0  32.0    Male      Bachelor's  Software Engineer   
1           1  28.0  Female        Master's       Data Analyst   
2           2  45.0    Male             PhD     Senior Manager   
3           3  36.0  Female      Bachelor's    Sales Associate   
4           4  52.0    Male        Master's           Director   

   Years of Experience    Salary Country      Race  
0                  5.0   90000.0      UK     White  
1                  3.0   65000.0     USA  Hispanic  
2                 15.0  150000.0  Canada     White  
3                  7.0   60000.0     USA  Hispanic  
4                 20.0  200000.0     USA     Asian

In [16]:
df.describe().T

count           mean           std    min       25%  \
Unnamed: 0           6704.0    3351.500000   1935.422435    0.0   1675.75   
Age                  6702.0      33.620859      7.614633   21.0     28.00   
Years of Experience  6701.0       8.094687      6.059003    0.0      3.00   
Salary               6699.0  115326.964771  52786.183911  350.0  70000.00   

                          50%        75%       max  
Unnamed: 0             3351.5    5027.25    6703.0  
Age                      32.0      38.00      62.0  
Years of Experience       7.0      12.00      34.0  
Salary               115000.0  160000.00  250000.0

In [17]:
df.shape

(6704, 9)

## There are 09 features and 6704 records in our dataset.

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6704 entries, 0 to 6703
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           6704 non-null   int64  
 1   Age                  6702 non-null   float64
 2   Gender               6702 non-null   object 
 3   Education Level      6701 non-null   object 
 4   Job Title            6702 non-null   object 
 5   Years of Experience  6701 non-null   float64
 6   Salary               6699 non-null   float64
 7   Country              6704 non-null   object 
 8   Race                 6704 non-null   object 
dtypes: float64(3), int64(1), object(5)
memory usage: 471.5+ KB


## Checking Missing Values

In [19]:
df.isna().sum()

Unnamed: 0             0
Age                    2
Gender                 2
Education Level        3
Job Title              2
Years of Experience    3
Salary                 5
Country                0
Race                   0
dtype: int64

As we have few null values as compared to the total reocrd.So, we can drop the null values.

## Fill Missing Values

In [20]:
df=df.dropna()

In [21]:
df.isnull().sum()

Unnamed: 0             0
Age                    0
Gender                 0
Education Level        0
Job Title              0
Years of Experience    0
Salary                 0
Country                0
Race                   0
dtype: int64

Making sure that there will be no null values now in our DS.

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6698 entries, 0 to 6703
Data columns (total 9 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Unnamed: 0           6698 non-null   int64  
 1   Age                  6698 non-null   float64
 2   Gender               6698 non-null   object 
 3   Education Level      6698 non-null   object 
 4   Job Title            6698 non-null   object 
 5   Years of Experience  6698 non-null   float64
 6   Salary               6698 non-null   float64
 7   Country              6698 non-null   object 
 8   Race                 6698 non-null   object 
dtypes: float64(3), int64(1), object(5)
memory usage: 523.3+ KB


Exporting the CSV file after cleaning.

In [23]:
df.to_csv('Clean.csv')